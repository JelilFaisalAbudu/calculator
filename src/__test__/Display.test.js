import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Display from '../components/Display';

afterEach(cleanup);

describe('Display', () => {
  test('displays the current result', () => {
    render(<Display result="45" />);
    const displayResult = screen.getByText('45');
    expect(displayResult).toBeInTheDocument();
  });

  test('Render the display component  correctly', () => {
    const displayTree = renderer.create(<Display />).toJSON();
    expect(displayTree).toMatchSnapshot();
  });
});
