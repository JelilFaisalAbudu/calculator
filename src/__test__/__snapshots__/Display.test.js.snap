// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Render the display component  correctly 1`] = `
<div
  className="display"
>
  0
</div>
`;
